### PR TITLE
test: adds test for numeric as primary key

### PR DESCRIPTION
**Pull request by thiagotnunes**
Adds test for numeric as primary key.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java [hunk 1]
```diff
@@ -151,4 +155,26 @@ public void instanceNotFound() {
       assertThat(e.getResourceName()).isEqualTo(nonExistingInstanceId.getName());
     }
   }
+
+  @Test
+  public void numericPrimaryKey()
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java:160]: nit: `testNumericPrimaryKey()`

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java [hunk 2]
```diff
@@ -151,4 +155,26 @@ public void instanceNotFound() {
...
+      throws InterruptedException, ExecutionException, TimeoutException {
+    final String instanceId = env.getTestHelper().getInstanceId().getInstance();
+    final String databaseId = env.getTestHelper().getUniqueDatabaseId();
+    final DatabaseAdminClient databaseAdminClient =
+        env.getTestHelper().getClient().getDatabaseAdminClient();
+
+    try {
+      final OperationFuture<Database, CreateDatabaseMetadata> op =
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java:168]: nit: op => operation

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java [hunk 3]
```diff
@@ -151,4 +155,26 @@ public void instanceNotFound() {
...
+          databaseAdminClient.createDatabase(
+              instanceId,
+              databaseId,
+              Collections.singletonList(
+                  "CREATE TABLE NumericTable (" + "Id NUMERIC NOT NULL" + ") PRIMARY KEY (Id)"));
+      final Database database = op.get(1, TimeUnit.MINUTES);
+      assertThat(database).isNotNull();
+    } finally {
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java:176]: Could we also add the following to this test:
1. Write a couple of values using mutations.
2. Read these back using a `Read` operation, as those require the user to specify the key(s) to read. That ensures that reading using numeric keys work.
3. Delete some of the values using mutations, as those require the user to specify the key(s) to delete.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java [hunk 4]
```diff
@@ -151,4 +155,26 @@ public void instanceNotFound() {
...
+        env.getTestHelper().getClient().getDatabaseAdminClient();
+
+    try {
+      final OperationFuture<Database, CreateDatabaseMetadata> op =
+          databaseAdminClient.createDatabase(
+              instanceId,
+              databaseId,
+              Collections.singletonList(
+                  "CREATE TABLE NumericTable (" + "Id NUMERIC NOT NULL" + ") PRIMARY KEY (Id)"));
+      final Database database = op.get(1, TimeUnit.MINUTES);
```
[the remark this answers is not in the document]
Comment (thiagotnunes) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java]: Fixed (10 minutes now)

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java (hunk 1 above)
Comment (thiagotnunes) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java]: Fixed

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java (hunk 2 above)
Comment (thiagotnunes) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java]: Fixed

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java (hunk 3 above)
Comment (thiagotnunes) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java:228]: Added these assertions to the test

**Review by olavloite — APPROVED** [review]

**Review by skuruppu — APPROVED** [review]

**Comment by olavloite** [comment]
@thiagotnunes Is there a specific reason that this has not yet been merged?

**Comment by thiagotnunes** [comment]
@olavloite yes, this is currently not supported by the backend yet.

**Pull request merged**